### PR TITLE
Implementing update

### PR DESCRIPTION
**Pull request by alejandroEsc**
currently working on the update part of the actuator, still a work in progress.

### CODE REVIEW

File: clusterctl/examples/ssh/bootstrap_scripts/node_teardown_ubuntu_16.04.template [hunk 1]
```diff
@@ -3,17 +3,16 @@
           (
           ARCH=amd64
 
-          mv -f /var/log/teardown.log /var/log/teardown.log.$(date +%Y-%m-%d-%H-%M)
+          sudo mv -f /var/log/teardown.log /var/log/teardown.log.$(date +%Y-%m-%d-%H-%M)
 
-          KUBECONFIG=/etc/kubernetes/admin.conf
```
Comment (davidewatson) [inline clusterctl/examples/ssh/bootstrap_scripts/node_teardown_ubuntu_16.04.template]: Is there a reason for this? The `kubectl` commands below are less readable this way.

Comment (alejandroEsc) [reply clusterctl/examples/ssh/bootstrap_scripts/node_teardown_ubuntu_16.04.template]: Main reasons: no where else do we do this, so if we move this route, lets apply it everywhere. Less important is that it confounds me during debug the env variables sent over when a command is sent with a script.

**Comment by alejandroEsc** [comment]
Rebased due to merging conflict

File: cloud/ssh/actuators/machine/setupconfigs.go [hunk 2]
```diff
@@ -62,9 +62,11 @@ func (v *ValidMachineConfigItems) matchMachineSetupConfig(params *MachineParams)
 		if !reflect.DeepEqual(paramRoles, validRoles) {
 			continue
 		}
-		if params.Versions != validParams.Versions {
-			continue
-		}
+		// AE: TODO ! Should we validate over versions? I would think not? otherwise we would need to change this
```
Comment (davidewatson) [inline cloud/ssh/actuators/machine/setupconfigs.go]: Should we predefine metadata for a few different k8s versions (e.g. 1.10.34, 1.10.6, 1.11.1) and leave this continue?

File: cloud/ssh/actuators/machine/actuator.go [hunk 3]
```diff
@@ -228,9 +231,71 @@ func (a *Actuator) Delete(c *clusterv1.Cluster, m *clusterv1.Machine) error {
 }
 
 // Update updates a machine and is invoked by the Machine Controller
-func (a *Actuator) Update(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
-	glog.Infof("Updating machine %v for cluster %v.", machine.Name, cluster.Name)
-	return fmt.Errorf("TODO: Not yet implemented")
+func (a *Actuator) Update(cluster *clusterv1.Cluster, goalMachine *clusterv1.Machine) error {
```
[the remark this answers is not in the document]
Comment (alejandroEsc) [reply cloud/ssh/actuators/machine/actuator.go]: done.

File: cloud/ssh/actuators/machine/actuator.go [hunk 4]
```diff
@@ -228,9 +231,71 @@ func (a *Actuator) Delete(c *clusterv1.Cluster, m *clusterv1.Machine) error {
...
+	goalVersions := goalMachine.Spec.Versions
+
+	if goalVersions.ControlPlane == currentVersionInfo.ControlPlane && goalVersions.Kubelet == currentVersionInfo.Kubelet {
+		glog.Infof("No updating required for Machine %s of cluster %s: ", goalMachine.Name, cluster.Name)
+		return nil
+	}
+
+	currentMachine.Spec.Versions = *currentVersionInfo
+
+	if util.IsMaster(currentMachine) {
```
[the remark this answers is not in the document]
Comment (alejandroEsc) [reply cloud/ssh/actuators/machine/actuator.go]: We probably should attempt this as a possibility. But would recommend that we do so in another branch/PR if we find the results are positive.

[the remark this answers is not in the document]
Comment (davidewatson) [reply cloud/ssh/actuators/machine/actuator.go]: Makes sense.

File: cloud/ssh/actuators/machine/actuator_helper.go [hunk 5]
```diff
@@ -2,17 +2,20 @@ package machine
 
 import (
 	"errors"
+	"fmt"
```
Comment (davidewatson) [inline cloud/ssh/actuators/machine/actuator_helper.go]: > Imports are organized in groups, with blank lines between them. The standard library packages are always in the first group. - https://github.com/golang/go/wiki/CodeReviewComments#imports

And I yes I realize there were already two `goimports` commits...

**Pull request merged**